posts/2024-09-19-Project-Goals-Sep-Update.md

```diff
@@ -0,0 +1,107 @@
+
+### Async Rust Parity (tracked in [#105])
+
+[#105]: https://github.com/rust-lang/rust-project-goals/issues/105
+[#129629]: https://github.com/rust-lang/rust/issues/129629
+
+We are generally on track with our marquee features: 
+
+1. Support for *async closures* is available on Nightly and the lang team arrived at a tentative consensus to keep the existing syntax (written rationale and formal decision are in progress). We issued a [call for testing](https://blog.rust-lang.org/inside-rust/2024/08/09/async-closures-call-for-testing.html) as well which has so far uncovered no issues.
+2. Partial support for *return-type notation* is available on Nightly with the remainder under review. 
+
+In addition, *dynamic dispatch for async functions* and *experimental async drop work* both made implementation progress. Async WG reorganization has made no progress. 
+
+[Read the full details on the tracking issue.](https://github.com/rust-lang/rust-project-goals/issues/105#issuecomment-2361474377)
+
+### Stabilize features needed by Rust for Linux (tracked in [#116])
+
+[#116]: https://github.com/rust-lang/rust-project-goals/issues/116
+
+We have stabilized [extended `offset_of` syntax](https://github.com/rust-lang/rust/pull/128284) and [agreed to stabilize Pointers to Statics in Constants](https://github.com/rust-lang/rust/issues/128183). Credit to @dingxiangfei2009 for driving these forward. 💜
+
+Implementation work proceeds for [arbitrary self types v2](https://github.com/rust-lang/rust/issues/44874#issuecomment-2314739657), [derive smart](https://github.com/rust-lang/rust/pull/125048) [pointer](https://github.com/rust-lang/rust/pull/129467), and [sanitizer support](https://github.com/rust-lang/rust/pull/128348).
+
+RFL on Rust CI is implemented but still waiting on documented policy. The first breakage was detected (and fixed) in [#129416](https://github.com/rust-lang/rust/pull/129416). This is the mechanism working as intended, although it would also be useful to better define what to do when breakage occurs.
+
+## Selected updates
+
+### Begin resolving cargo-semver-checks blockers for merging into cargo (tracked in [#104])
+
+[#104]: https://github.com/rust-lang/rust-project-goals/issues/104
+
+*@obi1kenobi has been working on laying the groundwork to enable manifest linting in their project. They have set up the ability to test how CLI invocations are interpreted internally, and can now snapshot the output of any CLI invocation over a given workspace. They have also designed the expansion of the CLI and the necessary Trustfall schema changes to support manifest linting. As of the latest update, they have a working prototype of manifest querying, which enables SemVer lints such as detecting the accidental removal of features between releases. This work is not blocked on anything, and while there are no immediate opportunities to contribute, they indicate there will be some in future updates.*
```

Comment:
Just my opinion, large blocks of italics can be difficult to read. Would it be possible to leave these as normal text? I'm not sure if you had some intention here.

Reply:
No problem, I'll change from italics. Not sure why I did that.